src/javascript/serverSettings/backgroundJobs/BackgroundJobsTable.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import {useTable, useSortBy} from 'react-table';
+import {
+    Loader,
+    Table,
+    TableBody,
+    TableBodyCell,
+    TableHead,
+    TableHeadCell,
+    TablePagination,
+    TableRow, Typography
+} from '@jahia/moonstone';
+import {renderSortIndicator} from './utils';
+import PropTypes from 'prop-types';
+import {useTranslation} from 'react-i18next';
+
+const NothingToDisplay = () => {
+    const {t} = useTranslation('serverSettings');
+    return (
+        <div style={{
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'center',
+            height: 100
+        }}
+        >
+            <Typography>{t('backgroundJobs.nothingToDisplay')}</Typography>
+        </div>
+    );
+};
+
+const BackgroundJobsTable = ({tableProps, paginationProps, loading, ...props}) => {
+    const {
+        limit,
+        setLimit,
+        totalCount,
+        currentPage,
+        setPage
+    } = paginationProps;
+
+    const {
+        getTableProps,
+        getTableBodyProps,
+        headerGroups,
+        rows,
+        prepareRow
+    } = useTable(tableProps, useSortBy);
+
+    if (loading) {
+        return (
+            <div style={{
+                display: 'flex',
+                alignItems: 'center',
+                justifyContent: 'center',
+                height: 300
+            }}
+            >
+                <Loader size="big"/>
+            </div>
+        );
+    }
+
+    const testId = props['data-testid'] ?? 'background-jobs-table';
```

Comment:
General comment: Would prefer to stick with the more commonly used test selectors we currently have: `data-sel-role` just for consistency. (I see it used in several places in the PR)